analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -18,58 +18,87 @@
-    }
+                }
+            },
+            SyntaxKinds);
+
+    protected virtual bool IsExcludedFromBeingExamined(SonarSyntaxNodeReportingContext context) =>
+        context.ContainingSymbol.Kind != SymbolKind.NamedType;
+
+    protected static bool HaveSameParameters<TSyntax>(SemanticModel model, SeparatedSyntaxList<TSyntax>? leftParameters, SeparatedSyntaxList<TSyntax>? rightParameters)
+        where TSyntax : SyntaxNode =>
+        (leftParameters is null && rightParameters is null) // In VB.Net the parameter list can be omitted
+        || (leftParameters?.Count == rightParameters?.Count && HaveSameParameterLists(model, leftParameters.Value, rightParameters.Value));
+
+    private static bool HaveSameParameterLists<TSyntax>(SemanticModel model,
+                                                        SeparatedSyntaxList<TSyntax> leftParameters,
+                                                        SeparatedSyntaxList<TSyntax> rightParameters) where TSyntax : SyntaxNode =>
+        leftParameters.Zip(rightParameters, (left, right) => left.IsEquivalentTo(right)).All(x => x)                    // Perf: Syntactic equivalence for all parameters first
+        && leftParameters.Zip(rightParameters, (left, right) => HaveSameParameterType(model, left, right)).All(x => x); // Also make sure the parameter types are same
+
+    private static bool HaveSameParameterType(SemanticModel model, SyntaxNode left, SyntaxNode right) =>
+        model.GetDeclaredSymbol(left) is IParameterSymbol { Type: { } leftParameterType }
+        && model.GetDeclaredSymbol(right) is IParameterSymbol { Type: { } rightParameterType }
+        && TypesAreSame(leftParameterType, rightParameterType);
+
+    private static bool TypesAreSame(ITypeSymbol leftParameterType, ITypeSymbol rightParameterType) =>
+        leftParameterType.Equals(rightParameterType) // M1(int x) <-> M2(int x)
+        || TypesAreSameTypeParameters(leftParameterType, rightParameterType) // M1<T>(T x) where T: class <-> M2<T>(T x) where T: class
+                                                                             // T of M1 is a different symbol than T of M2, but if they have the same constraints, they can be interchanged
+        || TypesAreSameGenericType(leftParameterType, rightParameterType); // M1<T>(IEnumerable<T> x) where T: class <-> M2<T>(IEnumerable<T> x) where T: class
+
+    private static bool TypesAreSameGenericType(ITypeSymbol leftParameterType, ITypeSymbol rightParameterType) =>
+        leftParameterType is INamedTypeSymbol { IsGenericType: true } namedTypeLeft
+        && rightParameterType is INamedTypeSymbol { IsGenericType: true } namedTypeRight
+        && namedTypeLeft.TypeParameters.Length == namedTypeRight.TypeParameters.Length
+        && namedTypeLeft.TypeParameters.Zip(namedTypeRight.TypeParameters, (left, right) => TypesAreSame(left, right)).All(x => x);
```

Comment:
This pattern is used multiple times in this class. We could write an overload to [SequenceEqual](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=net-8.0) that takes a lambda and does the code you do here under the hood (or create a `IEqualityComparer` on the fly). WDYT?

Reply:
Even better: There is already such an extension method.